includes/Admin.php

```diff
@@ -110,6 +110,24 @@ public function __construct() {
 		add_filter( 'gettext', array( $this, 'change_custom_taxonomy_tip' ), 20, 2 );
 	}
 
+	/**
+	 * __get method for backward compatibility.
+	 *
+	 * @param string $key property name
+	 * @return mixed
+	 * @since x.x.x
+	 */
+	public function __get( $key ) {
+		// Add warning for private properties.
+		if ( 'product_sets' === $key ) {
+			/* translators: %s property name. */
+			_doing_it_wrong( __FUNCTION__, sprintf( esc_html__( 'The %s property is private and should not be accessed outside its class.', 'facebook-for-woocommerce' ), esc_html( $key ) ), 'x.x.x' );
```

Comment:
A small nitpick: `product_sets` is a `protected.`

Reply:
Nice catch. I have fixed it in the [recent commit](https://github.com/woocommerce/facebook-for-woocommerce/pull/2619/commits/cd911a8ac92c716106bd5886a92c550caa797f92).